instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb

```diff
@@ -24,24 +32,39 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
           !defined?(::ActiveSupport::Notifications).nil? && gem_version >= MINIMUM_VERSION
         end
 
+        def subscribe
+          SUBSCRIPTIONS.each do |subscription_name|
+            OpenTelemetry.logger.debug("Subscribing to #{subscription_name} notifications with #{_tracer}")
+            OpenTelemetry::Instrumentation::ActiveSupport.subscribe(_tracer, subscription_name, default_attribute_transformer)
+          end
+        end
+
         private
 
+        def _tracer
+          self.class.instance.tracer
+        end
+
         def gem_version
           Gem::Version.new(::ActiveModel::Serializer::VERSION)
         end
 
-        def require_dependencies
-          require_relative 'event_handler'
+        def install_active_support_instrumenation
+          OpenTelemetry::Instrumentation::ActiveSupport::Instrumentation.instance.install({})
         end
 
-        def register_event_handler
-          ::ActiveSupport::Notifications.subscribe(event_name) do |_name, start, finish, _id, payload|
-            EventHandler.handle(start, finish, payload)
-          end
+        def require_dependencies
+          require_relative 'railtie'
         end
 
-        def event_name
-          'render.active_model_serializers'
+        def default_attribute_transformer
+          lambda { |payload|
```

Comment:
Bike shed time...

Is `serializer` a specific enough namespace not to collide with other instrumentations or sematic conventions?

Is `format` similar to a MIME type?

Reply:
These are probably not the greatest attribute names. In this PR, I was attempting to only fix the means by which the instrumentation subscribed so that current span for activity occurring within the event would be accurate. If there are semantic conventions to address in the attribute names, I propose we figure out what better attribute names could be used on a separate issue.